ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -85,6 +92,24 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 
 	logUnsupportedConfigFields(context.Project)
 
+	ecsParams, err := readECSParams(ecsParamsFileName)
+	if err != nil {
+		return nil, err
+	}
+
+	var networkMode string
+	var ecsContainerDefs ContainerDefs
+
+	if ecsParams != nil {
```

Comment:
I should have asked about this in the last PR, but why is `readECSParams()` allowed to return nil, nil? Your code implies that ecsParams could be nil, and the error could be nil, and this is not an error condition. Why?

Reply:
Because you might not have an `ecs-params.yml` file, in which case reading it shouldn't return anything and shouldn't be an error.

Reply:
Ah I see. This makes me think though that may be the processing of the ECS Params file should be moved to its own function. I think that might make the code easier to read. This is just a suggestion though.

Reply:
The processing of the file is already in its own function. It's called `readECSParams`.